File: JULY-2025/30-07-2025/p2419.swift
class Solution {
    func longestSubarray(_ nums: [Int]) -> Int {
        var maxNum = nums.max(), result = 1, current = 0
        for num in nums {
            if num == maxNum {
                current += 1
            } else {
                result = max(result, current)
                current = 0
            }
        }
        return max(result, current)
    }
}

File: JULY-2025/30-07-2025/p2419.kt
class Solution {
    fun longestSubarray(nums: IntArray): Int {

        val maxvalue = nums.maxOrNull()?:0
        var current =0
        var longest =0

        for(num in nums){
            if(num==maxvalue){
                current++
                longest = maxOf(current,longest)
            }else{
                current =0
            }
        }
        return longest
        
    }
}